backend/src/db-utils.ts

```diff
@@ -1,79 +1,193 @@
+): Promise<T | undefined> => {
+  const result = await query<T, Args>(args, options);
   return result[0];
 };
 
-const queryRows = async (args, options) => {
-  return await query(args, options || {});
+const queryRows = async <
+  T extends object,
+  Args extends Replacements = undefined
+>(
+  args: QueryArgs<Args>,
+  options: QueryOptions
+): Promise<T[]> => {
+  return await query<T, Args>(args, options);
 };
 
-const queryGenesisAccountCount = async () => {
-  return await querySingleRow(
+const queryGenesisAccountCount = async (): Promise<{ count: string }> => {
+  const result = await querySingleRow<{ count: string }>(
     [
       `SELECT
         COUNT(*)
       FROM accounts
       WHERE created_by_receipt_id IS NULL`,
     ],
-    { dataSource: DS_INDEXER_BACKEND }
+    { dataSource: DataSource.Indexer }
   );
+  if (!result) {
+    throw new Error("No accounts count!");
+  }
+  return result;
+};
+
+// model of "blocks" table in indexer DB
+type BlockModel = {
+  block_height: string;
+  block_hash: string;
+  prev_block_hash: string;
+  block_timestamp: string;
+  total_supply: string;
+  gas_price: string;
+  author_account_id: string;
+};
+
+// model of "nodes" table in telemetry DB
+type NodeModel = {
+  ip_address: string;
+  moniker: string;
+  account_id: string;
+  node_id: string;
+  last_seen: Date;
+  last_height: string;
+  agent_name: string;
+  agent_version: string;
+  agent_build: string;
+  peer_count: string;
+  is_validator: boolean;
+  last_hash: string;
+  signature: string;
+  status: string;
+  latitude: string | null;
+  longitude: string | null;
+  city: string | null;
+};
+
+type GenericNodeModelProps =
+  | "ip_address"
+  | "account_id"
+  | "node_id"
+  | "last_seen"
+  | "last_height"
+  | "agent_name"
+  | "agent_version"
+  | "agent_build"
+  | "status"
+  | "latitude"
+  | "longitude"
+  | "city";
+
+export type OnlineNode = {
+  accountId: string;
+  ipAddress: string;
+  nodeId: string;
+  lastSeen: Date;
+  lastHeight: number;
+  status: string;
+  agentName: string;
+  agentVersion: string;
+  agentBuild: string;
+  latitude: string | null;
+  longitude: string | null;
+  city: string | null;
+};
+
+export type StakingNodeWithTelemetryInfo = StakingNode & {
+  nodeInfo?: OnlineNode;
 };
 
 // query for node information
-const extendWithTelemetryInfo = async (nodes) => {
+const extendWithTelemetryInfo = async (
+  nodes: StakingNode[]
+): Promise<StakingNodeWithTelemetryInfo[]> => {
   const accountArray = nodes.map((node) => node.account_id);
-  let nodesInfo = await queryRows(
+  let nodesInfo = await queryRows<
+    Pick<NodeModel, GenericNodeModelProps>,
+    { accountArray: string[] }
+  >(
     [
       `SELECT ip_address, account_id, node_id,
         last_seen, last_height, status,
         agent_name, agent_version, agent_build,
         latitude, longitude, city
       FROM nodes
-      WHERE account_id IN (:accountArray)
+      WHERE account_id = ANY (:accountArray)
```

Comment:
This is probably fine in this query since there are not many records in the table, but It might introduce a performance hit in other queries, so beware of that.

Reply:
It will not, as `IN` and `= ANY` generate the same pg plan (but that's what stackoverflow says, let's check when we have metrics and can see the issue ourselves)